### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by pskucherov**

### CODE REVIEW

File: modules_in_project/finam/transaqConnector.js [hunk 1]
```diff
@@ -506,18 +480,18 @@ const functionCallbackHft = ffi.Callback(
     },
 );
 
-const functionCallbackNotHft = ffi.Callback(
```
Comment (pskucherov) [inline modules_in_project/finam/transaqConnector.js:509]: why was this deleted, but functionCallbackHft not?

File: modules_in_project/finam/transaqConnector.js [hunk 2]
```diff
@@ -545,29 +519,23 @@ function functionConnect(HftOrNot, callback) {
...
-            if (HftOrNot === 'Hft') {
-                return objectAccountsAndDll['afterInitialize'][HftOrNot].SetCallback(
-                    ffiCallback,
-                );
-            }
-            if (HftOrNot === 'NotHft') {
-                return objectAccountsAndDll['afterInitialize'][HftOrNot].SetCallback(
-                    ffiCallback,
-                );
-            }
```
Comment (pskucherov) [inline modules_in_project/finam/transaqConnector.js:569]: it was better

File: modules_in_project/finam/transaqConnector.js [hunk 3]
```diff
@@ -959,8 +927,11 @@ function functionGetHistory(queryObject) {
     const { count } = queryObject;
 
     // строка контракта меняется каждые 3 месяца, получить ее исходя из текущей даты
-    const contractString = functionActiveContractString();
```
Comment (pskucherov) [inline modules_in_project/finam/transaqConnector.js:962]: it was bug?

File: modules_in_project/finam/transaqConnector.js [hunk 4]
```diff
@@ -16,36 +16,10 @@ const arrayUnikStringCloseOpenPositions = []; // массив уникальны
...
-const isTransaqConnected = {
-    Hft: false,
-    NoHft: false,
-};
 
 // #region параметры подключения
 const board = 'FUT';
 
-// #region объект с логинами, паролями, dll файлами hft и НЕ hft, объектами dll
-const objectAccountsAndDll = {
```
Comment (pskucherov) [inline modules_in_project/finam/transaqConnector.js:28]: it must be global object for multiusers

File: server.js [hunk 5]
```diff
@@ -79,73 +79,88 @@ http
...
+                          
                                 console.log('pass expired');
                             }
 
                             if (message['server_status']) {
                                 if (message.server_status.connected === 'error' || message.server_status.connected === 'false') {
                                     // TODO: popup about connect error and redirect to login page
+                             
+                                  // redirect to login page
+                                  res.redirect('./login');
```
Comment (pskucherov) [inline server.js:137]: Is this the correct login page?
It's an Ajax request sent by opextrade. The redirect will happen not in UX, but in the API?

File: server.js [hunk 6]
```diff
@@ -160,26 +175,22 @@ http
...
-                        ].SendCommand(`<command id="${command}"/>`);
-                    } else if (arrayAnyWorldCommands.includes(command) === true) {
+                    if (arrayOneWorldCommands.includes(command)) {
+                        result = objectAccountsAndDll.afterInitialize[HftOrNot].SendCommand(`<command id="${command}"/>`);
+                    } 
+                    else if (arrayAnyWorldCommands.includes(command)) 
+                    {
                         if (command === 'change_pass') {
-                            if (!queryObject.oldpass || !queryObject.newpass) {
+                            if (!!queryObject.oldpass || !!queryObject.newpass) {
```
Comment (pskucherov) [inline server.js:159]: what?

File: server.js [hunk 7]
```diff
@@ -237,4 +241,4 @@ http
     })
     .listen(12345);
 
-// #endregion
+// #endregion
```
Comment (pskucherov) [inline server.js:225]: new line request

File: server.js [hunk 8]
```diff
@@ -79,73 +79,88 @@ http
...
-                            if (message.client && message.client.id) {
-                                transaqConnector.objectAccountsAndDll
-                                    .users[HftOrNot].Account = message.client;
-                                transaqConnector.objectAccountsAndDll
-                                    .users[HftOrNot].Account.clientId_1 = message.client.id;
-                            }
-
-                            if (message.messages && message.messages.message && message.messages.message.text === 'Password expired. Please change the password') {
+                            // set value if they exist
+                            transaqConnector.objectAccountsAndDll.users[HftOrNot].Account = !!message.client && !!message.client.id && message.client;
```
Comment (pskucherov) [inline server.js:125]: why write you `!!` ?

File: modules_in_project/finam/transaqConnector.js (hunk 4 above)
Comment (marufbelete) [reply modules_in_project/finam/transaqConnector.js:28]: yes I have corrected that in the new push

File: server.js (hunk 8 above)
Comment (marufbelete) [reply server.js:125]: it was just to convert truthy value in to boolean

File: server.js (hunk 5 above)
Comment (marufbelete) [reply server.js:137]: ok I didn't know that I will correct

File: server.js (hunk 6 above)
Comment (marufbelete) [reply server.js:159]: ohh sorry that was mistake

File: server.js (hunk 7 above)
Comment (marufbelete) [reply server.js:225]: I did or because we are just using or condition inside if statement  if (!!queryObject.oldpass || !!queryObject.newpass) {

File: modules_in_project/finam/transaqConnector.js (hunk 3 above)
Comment (marufbelete) [reply modules_in_project/finam/transaqConnector.js:962]: I have replaced this with async/await in the new file

File: modules_in_project/finam/transaqConnector.js (hunk 2 above)
Comment (marufbelete) [reply modules_in_project/finam/transaqConnector.js:569]: corrected in the new file

File: modules_in_project/finam/transaqConnector.js (hunk 1 above)
Comment (marufbelete) [reply modules_in_project/finam/transaqConnector.js:509]: corrected in new file
